src/base/name/chain.rs

```diff
@@ -42,7 +43,7 @@ pub struct Chain<L, R> {
 impl<L: ToLabelIter, R: ToLabelIter> Chain<L, R> {
     /// Creates a new chain from a first and second name.
     pub(super) fn new(left: L, right: R) -> Result<Self, LongChainError> {
-        if left.compose_len() + right.compose_len() > 255 {
+        if left.compose_len() + right.compose_len() > 255 { // TODO can't infer a specific type for Dname here
```

Comment:
I think you can use `Dname::MAX_LENGTH` here.

Or, technically, this should be 254 if `R` doesn’t end with the root label so that the root can be appended without errors.